src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx

```diff
@@ -53,8 +53,14 @@ class SavedQueryButtonGroup extends React.Component<Props> {
       };
     }
 
+    if (savedQueriesLoading) {
+      return prevState;
+    }
+
+    const savedEventView = EventView.fromSavedQuery(savedQuery);
+
     // Switching from a SavedQuery to another SavedQuery
-    if (queryId !== nextEventView.id) {
+    if (savedEventView.id !== nextEventView.id) {
```

Comment:
@leedongwei this one is slightly subtle. I've opted to use `savedEventView.id` to make it more explicit.